### ISSUE TITLE
The online editor converts tabs to spaces

### ISSUE THREAD

**Issue by jcbollinger** [koalaman/shellcheck#2085]
#### For bugs
- Rule Id SC1040
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

cat <<-EOF
        tab-indented text
        EOF
```

#### Here's what shellcheck currently says:
```
Line 3	SC1009: The mentioned syntax error was in this simple command.
Line 3	SC1073: Couldn't parse this here document. Fix to allow more checks.
Line 5	SC1040: When using <<-, you can only indent with tabs.
Line 6	SC1072: Here document was not correctly terminated. Fix any mentioned problems and try again.
```
#### Here's what I wanted or expected to see:

Shellcheck.net should not have reported issue SC1040 because the original script cut & pasted into the editor was properly tab-indented.  Apparently, however, the online editor automatically converts tabs into spaces.  What's more, the editor disguises that it has done so, in that it makes the arrow and backspace keys treat runs of 8 spaces as individual tab-like units (even when they are typed in as runs of spaces).

I am confident that it is the online editor at Shellcheck.net that is performing the unwanted conversion, because I can successfully cut & paste tabs into some other online editors, such as Stack Exchange's.

This behavior of the editor makes it difficult to use Shellcheck.net to check scripts that use indented heredocs, and it makes Shellcheck.net useless for checking such heredocs themselves: they must be unindented to get past SC1040, and SC1040 cannot simply be ignored because it is a syntax issue, preventing any semantic checks from being performed.

**Comment by jcbollinger** [comment]
Hmm. Having experimented a little more, it looks like the problem was on my side after all.  The editor still does weird things with runs of spaces, but that's not much of an issue.
